arctic/serialization/numpy_arrays.py

```diff
@@ -218,16 +222,19 @@ def deserialize(self, data, columns=None):
             if index:
                 columns = columns[:]
                 columns.extend(meta[INDEX])
-            if len(columns) > len(set(columns)):
-                raise Exception("Duplicate columns specified, cannot de-serialize")
+                columns = list(set(columns))
 
         if not isinstance(data, list):
             df = self.converter.objify(data, columns)
         else:
-            df = pd.concat([self.converter.objify(d, columns) for d in data], ignore_index=not index)
+            dfs = []
+            while len(data):
```

Comment:
is there a reason you modify the loop control variable in place during iteration? Is this for memory savings?

Reply:
You save having two copies of the data in memory. Will attach memory profile.

Reply:
Attached.